src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -200,5 +202,23 @@ public IEnumerable<TagHelperRequiredAttributeDescriptor> RequiredAttributes
         /// tag helper.
         /// </summary>
         public TagHelperDesignTimeDescriptor DesignTimeDescriptor { get; set; }
+
+        public Dictionary<string, string> PropertyBag
```

Comment:
Hmm, thinking we might want to make this lazy actually. In the design time case the laziness wont matter but at runtime it would. (This would remove the pranav points comment above).